src/main/java/gomedic/logic/parser/DateParser.java

```diff
@@ -0,0 +1,100 @@
+package gomedic.logic.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+
+/**
+ * Class that handles formatting the date.
+ * It handles 3 different date formats:
+ * dd/MM/yyyy HH:mm (e.g. 15/09/2022 13:00)
+ * dd-MM-yyyy HH:mm (e.g. 15-09-2022 13:00)
+ * yyyy-MM-dd-HH-mm (e.g. 2022-09-15-13-00)
+ */
+public class DateParser {
+
+    /**
+     * Checks if input String is a valid time and
+     * follows one of the 2 time formats that GoMedic allows:
+     * 13:00
+     * 13-00
+     */
+    public static boolean isValidTime(String time) {
+
+        String[] timeArr;
+
+        if (time.charAt(2) == ':') {
+            timeArr = time.split(":");
+        } else if (time.charAt(2) == '-') {
+            timeArr = time.split("-");
+        } else {
+            return false;
+        }
+
+        int hour;
+        int minute;
+
+        try {
+            hour = Integer.parseInt(timeArr[0]);
+            minute = Integer.parseInt(timeArr[1]);
+        } catch (NumberFormatException e) {
+            System.out.println("Invalid time");
+            return false;
+        }
+
+        boolean isValidHour = hour >= 0 && hour < 24;
+        boolean isValidMinute = minute >= 0 && minute < 60;
+
+        return (isValidHour && isValidMinute);
+    }
+
+    /**
+     * Formats the input datetime that was entered into a Date object
+     * These input formats are accepted:
+     * dd/MM/yyyy HH:mm (e.g. 15/09/2022 13:00)
+     * dd-MM-yyyy HH:mm (e.g. 15-09-2022 13:00)
+     * yyyy-MM-dd-HH-mm (e.g. 2022-09-15-13-00)
+     *
+     * @param dateLiteral date that was typed in by user
+     * @return Date object
+     */
+    public static LocalDateTime formatDate(String dateLiteral) {
+        DateTimeFormatter format;
+        if (dateLiteral.charAt(2) == '/') {
+            format = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");
+        } else if (dateLiteral.charAt(2) == '-') {
+            format = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+        } else {
+            format = DateTimeFormatter.ofPattern("yyyy-MM-dd-HH-mm");
+        }
+
+        try {
+            LocalDateTime dateFormatted = LocalDateTime.parse(dateLiteral, format);
+            return dateFormatted;
+        } catch (DateTimeParseException e) {
+            System.out.println("Incorrect date format");
+        }
+        return LocalDateTime.now();
```

Comment:
I think if we fail to format, its better to throw the error , instead of now(), to remove confusion where your time is not valid and suddenly overwritten with current time

Reply:
Okay, done